### ISSUE TITLE
[ENHANCEMENT] Bump KubeVirt to 1.1.0-150500.8.6.1

### ISSUE THREAD

**Issue by FrankYang0529** [harvester/harvester#4758]
**Is your enhancement related to a problem? Please describe.**
SLE 15 SP4 general ends on 31 Dec 2023. KubeVirt base image will change. We have to update to a new version.

The supported version for SLE 15 SP5 will be 1.0.0: registry.suse.com/suse/sles/15.5/virt-operator:1.0.0-150500.8.3.1

**Additional context**
https://www.suse.com/lifecycle/#suse-linux-enterprise-server-15

**Comment by TachunLin** [comment]
### Test Plan
1. Check can upgrade from v1.2.1 to v1.2-head until v1.3.0-rc2 can work as expected 
2. Check on fresh install and upgrade path to v1.3.0 Harvester cluster contains the bumped KubeVirt image to
    - `registry.suse.com/suse/sles/15.5/virt-operator:1.1.0-150500.8.6.1`
3. Check the basic VM operation can work 
4. Check the basic Volume operation can work

**Comment by TachunLin** [comment]
Verifying on `v1.3.0-rc1` (24/01/25).

### Result

<details><summary>$\color{Grey}{\textsf{Pending}}$ Check can  **upgrade** from v1.2.1 to v1.2-head until v1.3.0-rc2 $~~$</summary>

 * Wait until v1.3.0-rc2 released 



 </details>

<details><summary>$\color{Grey}{\textsf{Pending}}$ KubeVirt bumped to 1.1.0-150500.8.6.1 after **upgrade** from v1.2.1 to v1.2-head until v1.3.0-rc2 $~~$</summary>

 * Wait until v1.3.0-rc2 released 



 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ KubeVirt bumped to 1.1.0-150500.8.6.1 after **fresh install** Harvester cluster $~~$</summary>

1. In fresh install Harvester v1.3.0-rc1, Kubevirt already bumped to `1.1.0-150500.8.6.1`

```
node1:~ # ctr -n k8s.io image ls | grep virt-operator
registry.suse.com/suse/sles/15.5/virt-operator:1.1.0-150500.8.6.1                                      application/vnd.docker.distribution.manifest.v2+json sha256:6a1b32dd2e1732c587c15f3673ea97317a6210735425d50072f1f146eed6a398 181.1 MiB linux/amd64 io.cri-containerd.image=managed
```

 </details>

<details><summary>$\color{red}{\textsf{Failed}}$ Basic Volume operation test plans $~~$</summary>

 1. $\color{green}{\textsf{PASS}}$ the following volumes backend automation test plans  $~~$

    - [x]    - test_1_volumes.py::test_create_volume_backing_image[json]
    - [x]    - test_1_volumes.py::test_create_volume_backing_image[yaml]
    - [x]    - test_1_volumes.py::TestVolumeWithVM::test_delete_volume_bounded_on_existing_vm
    - [x]    - test_1_volumes.py::TestVolumeWithVM::test_delete_volume_bounded_on_deleted_vm

2. Manual test plans
    - [x]   [Add/remove disk to Host config](https://harvester.github.io/tests/manual/volumes/1623-add-disk-to-host/)
    - [ ]    [Create Volume root disk blank Form with label](https://harvester.github.io/tests/manual/volumes/create-volume-root-disk-blank-form-label/) 
         - Failed with known issue: https://github.com/harvester/harvester/issues/5020
    - [x]  [Delete volume that is not attached to a VM (e2e_be_fe)](https://harvester.github.io/tests/manual/volumes/delete-volume-that-is-not-attached-to-vm/)
    - [x]   [Delete volume that was attached to VM but now is not (e2e_be_fe)](https://harvester.github.io/tests/manual/volumes/delete-volume-that-was-attached-to-vm-but-is-not-now/)
    - [x]   [Detach volume from virtual machine (e2e_fe)](https://harvester.github.io/tests/manual/volumes/detach-volume-from-virtual-machine/)
    - [x]  [Edit volume increase size via YAML (e2e_be)](https://harvester.github.io/tests/manual/volumes/edit-volume-increase-size-yaml/)
    - [x]  [Edit volume increase size via form (e2e_fe)](https://harvester.github.io/tests/manual/volumes/edit-volume-increase-size-form/)
    - [x] [Negative delete Volume that is in use (e2e_be)](https://harvester.github.io/tests/manual/volumes/negative-delete-volume-that-is-in-use/)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Basic VM operation test plans $~~$</summary>

1. $\color{green}{\textsf{PASS}}$ the following VM backend automation test plans  $~~$

    - [x]    - test_3_vm_functions.py::test_minimal_vm
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_pause
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_unpause
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_stop
    - [x]    - test_3_vm_functions.py::test_minimal_vm
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_start
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_restart
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_softreboot
    - [x]    - test_3_vm_functions.py::test_minimal_vm
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_migrate
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_delete
    - [x]    - test_3_vm_functions.py::test_create_stopped_vm
    - [x]    - test_3_vm_functions.py::TestVMResource::test_create_schedule_on_maximum[cpu]
    - [x]    - test_3_vm_functions.py::TestVMResource::test_create_schedule_on_maximum[memory]
    - [x]    - test_3_vm_functions.py::TestVMResource::test_update_schedule_on_maximum[cpu]
    - [x]    - test_3_vm_functions.py::TestVMResource::test_update_schedule_on_maximum[memory]
    - [x]    - test_3_vm_functions.py::test_minimal_vm
    - [x]    - test_3_vm_functions.py::TestVMResource::test_update_enable_usb
    - [x]    - test_3_vm_functions.py::TestVMResource::test_update_enable_user_data
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[cpu]
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[mem]
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[disk]
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[mem-and-cpu]
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[mem-cpu-and-disk]
    - [x]    - test_3_vm_functions.py::test_vm_with_bogus_vlan
    - [x]     - test_3_vm.py::test_multiple_migrations
    - [x]     - test_3_vm.py::test_migrate_vm_with_user_data
    - [x]     - test_3_vm.py::test_migrate_vm_with_multiple_volumes

1. $\color{red}{\textsf{FAILED}}$ the following VM backend automation test plans  $~~$

    - [x]    - test_3_vm_functions.py::TestVMOperations::test_abort_migrate
    - [x]    - test_3_vm_functions.py::test_update_vm_machine_type[pc_to_q35]
    - [x]    - test_3_vm_functions.py::test_update_vm_machine_type[q35_to_pc]


 </details>


### Test Information
* Test Environment: 3 nodes harvester on baremetal machines
* Harvester version: `v1.3.0-rc1` (24/01/25)

### Verify Steps
<details><summary>Check can upgrade from v1.2.1 to v1.2-head until v1.3.0-rc2  </summary>


</details>

<details><summary>Check the KubeVirt can be bumped to 1.1.0-150500.8.6.1 after upgrade complete or fresh installation </summary>


</details>


<details><summary> Basic volume operation test</summary>

1. Run backend automation tests `test_volume.py` on internal E2E Jenkins jobs
2. Run manual test plan to validate basic user operations

</details>

<details><summary> Basic vm operation test </summary>

1. Run backend automation tests `test_vm.py` and `test_vm_function.py` on internal E2E Jenkins jobs

</details>

**Comment by innobead** [comment]
@khushboo-rancher We need to ensure verifying ready-for-test issues is always priority first, so that we will be able to capture issues earlier. 

In this issue, the PR was merged last December, but we are just testing it right now. There is an issue already found by engineering at https://github.com/harvester/harvester/issues/5076#issuecomment-1927301846 before QA.

This is a general rule, so let's make this apply to the team. Thanks.

cc @harvester/qa

**Comment by lanfon72** [comment]
Issue about update machine type: https://github.com/harvester/harvester/issues/5005

This issue moved to **ready for testing** in _Feb 1_, I think it is the mainly reason QA didn't notice it.
![image](https://github.com/harvester/harvester/assets/5169694/319bc572-6dbc-4539-8e77-b5a68153b8dd)

QA usually will not test upgrade before RC release, and our automation upgrade testing didn't contains VM migration while upgrading, that's why we didn't catch the bug which RD filed.

In `v1.3.0`, upgrade testing is deferred to **RC2** (QA also delay upgrade testing in `v1.3.0-rc1`) because we are bumping Rancher's version[^1], and will not support `v1.2.1` upgrade to `v1.3.0`.

[^1]: https://github.com/harvester/harvester-installer/pull/643

**Comment by innobead** [comment]
@lanfon72 Good clarification. Updating the state of issues should be improved, so we can let QA get involved earlier.

**Comment by TachunLin** [comment]
For the issue verification on `v1.3.0-rc1`, we executed backend automation test to validate basic function of virtual machine and volume. And also run some manual test plans on volume function.

Most of the automation [test result](https://github.com/harvester/harvester/issues/4758#issuecomment-1925613014) for basic vm and volume function looks good. 

During the manual testing of volume function, we encounter the known issue #5020 of volume would display `not ready` state when we create a new source volume without binding to a VM.

And we also have the known issue #5076 for vm can't migrate during upgrade from v1.2.1 to v1.3.0-rc1.

Thanks and according to above [comment](https://github.com/harvester/harvester/issues/4758#issuecomment-1930594384), we deferred the upgrade test until `v1.3.0-rc2` release since we have to bump the Rancher release. 

And follow the upgrade path from v1.2.1 to `v1.2-head` until `v1.3.0-rc2` for this issue.  

Next we will start testing the pending upgrade test item on `v1.3.0-rc2` and rerun all the automation and manual test plans in the following second run verification

**Comment by TachunLin** [comment]
Verifying on `v1.3-head` (24/02/21) and `v1.3.0-rc3`. 

### Result

<details><summary>$\color{green}{\textsf{PASS}}$ Check can  **upgrade** from v1.2.1 to v1.2-head until v1.3.0-head $~~$</summary>


* We can successfully upgrade to v1.3-head (`v1.3-50a00ad1-head`)
![image](https://github.com/harvester/harvester/assets/29251855/abb09dfa-ab19-4fc0-bb39-fb84f9003854)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ KubeVirt bumped to 1.1.0-150500.8.6.1 after **upgrade** from v1.2.1 to v1.2-head until v1.3-head $~~$</summary>

```
node1:~ # ctr -n k8s.io image ls | grep virt-operator
...
registry.suse.com/suse/sles/15.5/virt-operator:1.1.0-150500.8.6.1                                       application/vnd.docker.distribution.manifest.v2+json sha256:6a1b32dd2e1732c587c15f3673ea97317a6210735425d50072f1f146eed6a398 181.1 MiB linux/amd64 io.cri-containerd.image=managed
```

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ KubeVirt bumped to 1.1.0-150500.8.6.1 after **fresh install** Harvester cluster $~~$</summary>

  ```
  harvester-kmzrw:~ # ctr -n k8s.io image ls | grep virt-operator
  registry.suse.com/suse/sles/15.5/virt-operator:1.1.0-150500.8.6.1                                      application/vnd.docker.distribution.manifest.v2+json sha256:6a1b32dd2e1732c587c15f3673ea97317a6210735425d50072f1f146eed6a398 181.1 MiB linux/amd64 io.cri-containerd.image=managed
  ```

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Volume backend automation test $~~$</summary>

 * $\color{green}{\textsf{PASS}}$ the following volumes backend automation test plans  $~~$

    - [x]    - test_1_volumes.py::test_create_volume_backing_image[json]
    - [x]    - test_1_volumes.py::test_create_volume_backing_image[yaml]
    - [x]    - test_1_volumes.py::TestVolumeWithVM::test_delete_volume_bounded_on_existing_vm
    - [x]    - test_1_volumes.py::TestVolumeWithVM::test_delete_volume_bounded_on_deleted_vm

</details>

<details><summary>$\color{green}{\textsf{PASS}}$ Volume manual test plans $~~$</summary>

 * $\color{green}{\textsf{PASS}}$ the following volumes manual automation test plans  $~~$
    - [x]   [Create image from Volume(e2e_fe)](https://harvester.github.io/tests/manual/volumes/create-image-from-volume/)
    - [x]   [Create Volume root disk blank Form with label](https://harvester.github.io/tests/manual/volumes/create-volume-root-disk-blank-form-label/)
    - [x]  [Create volume root disk VM Image Form with label](https://harvester.github.io/tests/manual/volumes/create-volume-root-disk-vm-image-form-label/) 
    - [x]  [Create volume root disk VM Image Form(e2e_fe)](https://harvester.github.io/tests/manual/volumes/create-volume-root-disk-vm-image-form/)
    - [x]  [Delete volume that is not attached to a VM (e2e_be_fe)](https://harvester.github.io/tests/manual/volumes/delete-volume-that-is-not-attached-to-vm/) 
    - [x]   [Delete volume that was attached to VM but now is not](https://harvester.github.io/tests/manual/volumes/delete-volume-that-was-attached-to-vm-but-is-not-now/) 
    - [x]    [Detach volume from virtual machine (e2e_fe)](https://harvester.github.io/tests/manual/volumes/detach-volume-from-virtual-machine/)
    - [x]     [Edit volume increase size via form (e2e_fe) ](https://harvester.github.io/tests/manual/volumes/edit-volume-increase-size-form/)
    - [x]     [Edit volume increase size via YAML (e2e_be)](https://harvester.github.io/tests/manual/volumes/edit-volume-increase-size-yaml/) 
    - [x]      [Negative delete Volume that is in use (e2e_be)](https://harvester.github.io/tests/manual/volumes/negative-delete-volume-that-is-in-use/) 
    - [x]      [Validate volume shows as in use when attached (e2e_be)](https://harvester.github.io/tests/manual/volumes/validate-volume-shows-in-use-while-attached/)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ VM backend automation test $~~$</summary>

 * $\color{green}{\textsf{PASS}}$ the following VM backend automation test plans  $~~$

    - [x]    - test_3_vm_functions.py::test_minimal_vm
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_pause
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_unpause
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_stop
    - [x]    - test_3_vm_functions.py::test_minimal_vm
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_start
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_restart
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_softreboot
    - [x]    - test_3_vm_functions.py::test_minimal_vm
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_migrate
    - [x]    - test_3_vm_functions.py::TestVMOperations::test_delete
    - [x]    - test_3_vm_functions.py::test_create_stopped_vm
    - [x]    - test_3_vm_functions.py::TestVMResource::test_create_schedule_on_maximum[cpu]
    - [x]    - test_3_vm_functions.py::TestVMResource::test_create_schedule_on_maximum[memory]
    - [x]    - test_3_vm_functions.py::TestVMResource::test_update_schedule_on_maximum[cpu]
    - [x]    - test_3_vm_functions.py::TestVMResource::test_update_schedule_on_maximum[memory]
    - [x]    - test_3_vm_functions.py::test_minimal_vm
    - [x]    - test_3_vm_functions.py::TestVMResource::test_update_enable_usb
    - [x]    - test_3_vm_functions.py::TestVMResource::test_update_enable_user_data
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[cpu]
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[mem]
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[disk]
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[mem-and-cpu]
    - [x]    - test_3_vm_functions.py::test_create_vm_no_available_resources[mem-cpu-and-disk]
    - [x]    - test_3_vm_functions.py::test_vm_with_bogus_vlan
    - [x]     - test_3_vm.py::test_multiple_migrations
    - [x]     - test_3_vm.py::test_migrate_vm_with_user_data
    - [x]     - test_3_vm.py::test_migrate_vm_with_multiple_volumes

 </details>


<details><summary>$\color{red}{\textsf{FAIL}}$ VM manual test plans $~~$</summary>

 * $\color{blue}{\textsf{Ongoing}}$ the following VM backend automation test plans  $~~$

    - [x]   Basic VM operation: Create / Stop / Start / Restart / Delete 
    - [x]  [Chain VM templates and images](https://harvester.github.io/tests/manual/virtual-machines/760-chained-vm-templates/)
    - [x]  [Check VM creation required-fields](https://harvester.github.io/tests/manual/virtual-machines/1283-vm-creation-required-fields/)
    - [x]  [Clone VM that is turned on](https://harvester.github.io/tests/manual/virtual-machines/clone-vm-that-is-turned-on/) 
    - [x]   [Clone VM that is turned off](https://harvester.github.io/tests/manual/virtual-machines/clone-vm-that-is-turned-off/) 
    - [x]    Take vm backup 
    - [x]    Take vm snapshot
    - [x]    Restore vm backup to replace existing
    - [x]    Restore vm backup to create new 
    - [x]    Restore vm from snapshot
    - [x]    Clone vm with volume data
    - [x]    Migrate vm to other node 
    - [x]    Add volumes to existing vm
    - [x]    Edit yaml of vm
    - [x]    Create vm from node selector on specific node 
    - [x]    Create vm from node selector using node label
    - [x]     Edit the vm, change the cpu and memory size 
    - [x]     Edit the vm, increase the volume size 
    - [x]     [CPU overcommit on VM (e2e_fe)](https://harvester.github.io/tests/manual/virtual-machines/cpu_overcommit/)
    - [ ]      [Memory overcommit on VM](https://harvester.github.io/tests/manual/virtual-machines/memory_overcommit/)
  $\color{red}{\textsf{Failed}}$ Global memory over-commit not working $~~$
                
    ![image](https://github.com/harvester/harvester/assets/29251855/5fbeb26c-86e3-46df-ad50-07e75d4cee99)

    - [ ]      [Add a network to an existing VM with only 1 network](https://harvester.github.io/tests/manual/virtual-machines/add-a-network-to-an-existing-vm-with-only-1-network/)
    
       $\color{red}{\textsf{Failed}}$ Can't display the second NIC ip address on dashboard $~~$      
        ![image](https://github.com/harvester/harvester/assets/29251855/b841f970-19fa-464f-9f48-1d80db37b397)



    - [ ]      [Create a VM with 2 networks (e2e_be)](https://harvester.github.io/tests/manual/virtual-machines/add-a-network-to-an-existing-vm-with-only-1-network/)


      $\color{red}{\textsf{Failed}}$ Can't display NIC ip address on dashboard $~~$ 
      ![image](https://github.com/harvester/harvester/assets/29251855/b841f970-19fa-464f-9f48-1d80db37b397)


 </details>

### Test Information
* Test Environment: three and four nodes harvester on baremetal machines
* Harvester version: ` v1.3-50a00ad1-head` (24/02/21)

### Verify Steps
<details><summary>Check can upgrade from v1.2.1 to v1.2-head until v1.3-head  </summary>


</details>

<details><summary>Check the KubeVirt can be bumped to 1.1.0-150500.8.6.1 after upgrade complete or fresh installation </summary>


1. Execute command `ctr -n k8s.io image ls | grep virt-operator`


</details>


<details><summary> Basic volume operation test</summary>

1. Run backend automation tests `test_volume.py` on internal E2E Jenkins jobs
2. Run manual test plan to validate basic user operations

</details>

<details><summary> Basic vm operation test </summary>

1. Run backend automation tests `test_vm.py` and `test_vm_function.py` on internal E2E Jenkins jobs
2. Run manual test plan to validate basic user operations

</details>

**Comment by FrankYang0529** [comment]
> During the manual testing of volume function, we encounter the known issue #5020 of volume would display `not ready` state when we create a new source volume without binding to a VM.

From https://github.com/harvester/harvester/issues/5020#issuecomment-1911535650, #5020 is frontend issue and it's related to LH. I think we can skip it. Thank you.

**Comment by TachunLin** [comment]
Update the upgrade test status, we follow the upgrade path for this issue:
`v1.2.1` -> `v1.2-head` -> `v1.3-head`


#### Test environment and Prerequisite
* Test Environment: Three nodes harvester on remote kvm machines
* Harvester version: `v1.2.1` -> `v1.2-head` -> `v1.3-head`
* Rancher version: v2.8.1
* One RKE2 guest cluster v1.25.16 and VM is running

#### [PASS] Upgrade from v1.2.1 to latest v1.2-head (24/02/19)

* Before upgrade, we already have one RKE2 guest cluster VM running on Harvester

* We can successfully upgrade Harvester from v1.2.1 to v1.2-head  (v1.2-d542563e-head)
![image](https://github.com/harvester/harvester/assets/29251855/8006a427-dd2f-42dc-85f6-60856ed090f5)

* During the `upgrading system service` stage, it can upgrade KubeVirt to `1.1.0-150500.8.6.1`
![image](https://github.com/harvester/harvester/assets/29251855/ce5eb9e4-2a2e-4910-9bee-17e5aab20ab6)

* And also can upgrade the longhorn-manager 
![image](https://github.com/harvester/harvester/assets/29251855/e0496deb-828d-440f-878e-38d67cc5da1a)

* After upgraded to v1.2-head, we could find the VM already been shutdowned
![image](https://github.com/harvester/harvester/assets/29251855/1b9db963-f88a-41d3-9dfc-2c1dac4ccce4)



#### [Failed] Upgrade from v1.2-head to v1.3-head

Upgrade to v1.3-head **failed** when one VM is running. 

* The upgrade process would stuck in `Pre-draining` of the node host the VM (VM keeps in migrating)
![image](https://github.com/harvester/harvester/assets/29251855/028db571-0f10-4ec7-8aa9-91f8ae066995)


* Check the pre-drain pod log, you can find it keeps looping in `VM live-migration or shutdown` 

![image](https://github.com/harvester/harvester/assets/29251855/560b01e2-33af-47ca-99b7-2a1243e4e804)




```
Waiting for VM live-migration or shutdown...(1 left)                                                                                                       │
│ + '[' true ']'                                                                                                                                             │
│ ++ get_running_vm_count                                                                                                                                    │
│ ++ local count                                                                                                                                             │
│ +++ kubectl get vmi -A -l kubevirt.io/nodeName=harvester-node-1 -o yaml                                                                                    │
│ +++ yq e '.items[] | [select(.status.phase!="Succeeded")] | length'                                                                                        │
│ ++ count=1                                                                                                                                                 │
│ ++ echo 1                                                                                                                                                  │
│ + vm_count=1                                                                                                                                               │
│ + '[' 1 = 0 ']'                                                                                                                                            │
│ + echo 'Waiting for VM live-migration or shutdown...(1 left)'                                                                                              │
│ + sleep 5                                                                                                                                                  │
│ Waiting for VM live-migration or shutdown...(1 left)      
```

* During the upgrade we would encounter issue #5076 which cause VM failed to migrate to another node,

**Comment by TachunLin** [comment]
During the upgrade path from `v1.2.1` to `v1.2-head` until `v1.3-head`
According to the previous [test result](https://github.com/harvester/harvester/issues/4758#issuecomment-1953632139) 

1. With RKE2 guest cluster running on Harvester, we can upgrade from v1.2.1 to v1.2-head. 
2. After upgraded to v1.2-head, the VM would be automatically shutdown
3. We start the guest cluster VM again and proceed the upgrade to v1.3-head
4. The upgrade process to v1.3-head would stuck in `pre-draining` which would be blocked by issue #5076 

I would start over another upgrade without VM only confirm we can upgrade in a empty environment and especially the KubeVirt have been bumped to the expected version.

**Comment by TachunLin** [comment]
#### [PASS] Upgrade path v1.2.1 > v1.2-head > v1.3-head (without VM)
* We can successfully upgrade to v1.3-head (`v1.3-50a00ad1-head`)
![image](https://github.com/harvester/harvester/assets/29251855/abb09dfa-ab19-4fc0-bb39-fb84f9003854)

* After upgrade to v1.3-head, the KubeVirt bumped to `1.1.0-150500.8.6.1`
```
node1:~ # ctr -n k8s.io image ls | grep virt-operator
...
registry.suse.com/suse/sles/15.5/virt-operator:1.1.0-150500.8.6.1                                       application/vnd.docker.distribution.manifest.v2+json sha256:6a1b32dd2e1732c587c15f3673ea97317a6210735425d50072f1f146eed6a398 181.1 MiB linux/amd64 io.cri-containerd.image=managed
```




#### [PASS] Upgrade path v1.2.1 > v1.2-head > v1.3.0-rc3 (with VM shutdown)
* We can successfully upgrade to v1.3.0-rc3 
![image](https://github.com/harvester/harvester/assets/29251855/3295d37c-3a05-4da5-9ba1-799f448fb615)


* After upgrade to v1.3-head, the KubeVirt bumped to `1.1.0-150500.8.6.1`
```
harvester-node-2:~ # ctr -n k8s.io image ls | grep virt-operator
registry.suse.com/suse/sles/15.5/virt-operator:1.1.0-150500.8.6.1                                       application/vnd.docker.distribution.manifest.v2+json      sha256:6a1b32dd2e1732c587c15f3673ea97317a6210735425d50072f1f146eed6a398 181.1 MiB linux/amd64                                       io.cri-containerd.image=managed
```

#### Notice
Currently when upgrade from v1.2-head to v1.3.0-rc3, please use this [workaround](https://github.com/harvester/harvester/issues/3040#issuecomment-1298057127) to bypass the [upgrade version limitation](https://github.com/harvester/harvester/blob/1bda61b3c3ed6f1320eca1293d7aaeff721d00c2/package/upgrade-matrix.yaml#L2-L3) (Remove the current upgrade then apply the workaround)

**Comment by TachunLin** [comment]
While running the virtual machine manual test plan after upgraded to v1.3.0-rc3. 
We encounter the known issue #5005 for not able to create `pc` machine type vm. 

Double check the vm machine type in the following test scenarios, update the test result and impact

* After upgraded to v1.3.0-rc3: 
   - **Can't** create vm with the `pc` machine type, prompted the machine type not supported
   
      ![image](https://github.com/harvester/harvester/assets/29251855/2c0d2d54-5f48-408b-8af1-f3fea8f734f2)

   - Can create vm with the `qc35` machine type 
   - **Can't** cover the vm already have `qc35` machine type to `pc` type, prompted the same message
   
* Already have one `pc` type and one `qc35` type vm in v1.2.1 then upgrade to v1.3.0-rc3 
  - **Can't** start the `pc` type vm already exists in v1.2.1 
  
    ![image](https://github.com/harvester/harvester/assets/29251855/5ceadbf6-4a35-4d58-9c9f-7e9d3bb23279)

  - Can start the `qc35` type vm already exists in v1.2.1
  - **Can't** convert `qc35` type vm to `pc` type.

**Comment by TachunLin** [comment]
Complete the verification on `v1.3.0-rc3` and `v1.3-head`

According to the [test result](https://github.com/harvester/harvester/issues/4758#issuecomment-1945297059), we can summarize the following result:

1. After shutdown all vms (for #5076), 
We **can** successfully upgrade from `v1.2.1` to `v1.2-head` until `v1.3.0-rc3` 
2. After upgrade to `v1.2-head` and `v1.3.0-rc3`, the KubeVirt **can** bumped to `1.1.0-150500.8.6.1`
3. In a newly installed `v1.3.0-rc3` cluster, the KubeVirt **can** bumped to `1.1.0-150500.8.6.1`
4. **Pass** all volume backend automation test plans
5. **Pass** all volume manual test plans
6. **Pass** most of the virtual machine backend automation test, **except** the pc type releted test plans (for #5005)
7. Found some known and new issues while running the virtual machine manual test plans as the following:
    - #5005 (Known issue)
    - #5272 
    - #5266 
    - #5264 
    - #5276 

@bk201, @FrankYang0529 
May I confirm whether above issues are related to Kubevirt functionality
Since most of the basic features of virtual machine and volume are work as expected.
If they are not related to the scope of KubeVirt, I think we can close this enhancement ticket and track these issues separately. Thank you.

**Comment by bk201** [comment]
@TachunLin we can close this one and track those remaining with the issues you file.

**Comment by TachunLin** [comment]
Close this issue since we achieved most of the expected test result for bumping the Kubevirt and checked the basic volume and virtual machine functionalities. 

We would separately track the remaining issues afterward.
